File: ios/App/UITests/Pages/AuthPage.swift
import XCTest

/// Page object for authentication screens
class AuthPage {
    let app: XCUIApplication
    
    init(app: XCUIApplication) {
        self.app = app
    }
    
    // MARK: - Sign In View Elements
    
    var signInButton: XCUIElement {
        app.buttons["Sign in"]
    }
    
    var continueWithoutAccountButton: XCUIElement {
        app.buttons["Continue without account"]
    }
    
    var tallyLogo: XCUIElement {
        app.staticTexts["Tally"]
    }
    
    var tagline: XCUIElement {
        app.staticTexts["Track what matters"]
    }
    
    // MARK: - Clerk Auth Sheet Elements (when shown)
    
    var emailTextField: XCUIElement {
        // Clerk's email field in the auth sheet
        app.textFields.firstMatch
    }
    
    var passwordTextField: XCUIElement {
        // Clerk's password field
        app.secureTextFields.firstMatch
    }
    
    var continueButton: XCUIElement {
        app.buttons["Continue"]
    }
    
    // MARK: - Actions
    
    func tapSignIn() {
        signInButton.tap()
    }
    
    func tapContinueWithoutAccount() {
        continueWithoutAccountButton.tap()
    }
    
    func enterEmail(_ email: String) {
        guard emailTextField.waitForExistence(timeout: 5) else {
            XCTFail("Email field not found")
            return
        }
        emailTextField.tap()
        emailTextField.typeText(email)
    }
    
    func enterPassword(_ password: String) {
        guard passwordTextField.waitForExistence(timeout: 5) else {
            XCTFail("Password field not found")
            return
        }
        passwordTextField.tap()
        passwordTextField.typeText(password)
    }
    
    func tapContinue() {
        continueButton.tap()
    }
    
    // MARK: - Assertions
    
    func assertSignInViewIsVisible() {
        XCTAssertTrue(
            tallyLogo.waitForExistence(timeout: 10),
            "Tally logo should be visible on sign in screen"
        )
        XCTAssertTrue(signInButton.exists, "Sign in button should be visible")
    }
    
    func assertIsSignedOut() {
        // Verify we're on the sign in screen
        XCTAssertTrue(
            signInButton.waitForExistence(timeout: 10),
            "Should see sign in button when signed out"
        )
    }
}

File: tally-android/app/src/androidTest/java/com/tally/app/pages/AuthPage.kt
package com.tally.app.pages

import androidx.compose.ui.test.junit4.ComposeTestRule
import androidx.compose.ui.test.onNodeWithTag
import androidx.compose.ui.test.onNodeWithText
import androidx.compose.ui.test.onNodeWithContentDescription
import androidx.compose.ui.test.performClick
import androidx.compose.ui.test.assertIsDisplayed

/**
 * Page object for the Sign-in screen.
 */
class AuthPage(private val composeRule: ComposeTestRule) {
    
    // MARK: - Element Finders
    
    fun signInScreen() = composeRule.onNodeWithTag("sign_in_screen")
    
    fun signInButton() = composeRule.onNodeWithTag("sign_in_button")
    
    fun continueWithoutAccountButton() = composeRule.onNodeWithTag("continue_without_account_button")
    
    fun tallyLogo() = composeRule.onNodeWithContentDescription("Tally app logo")
    
    fun appTitle() = composeRule.onNodeWithTag("app_title")
    
    fun tagline() = composeRule.onNodeWithTag("app_tagline")
    
    fun localOnlyModeNote() = composeRule.onNodeWithText("Your data stays on this device in local-only mode.", substring = true)
    
    // MARK: - Actions
    
    fun tapSignIn() {
        signInButton().performClick()
    }
    
    fun tapContinueWithoutAccount() {
        continueWithoutAccountButton().performClick()
    }
    
    // MARK: - Assertions
    
    fun assertSignInViewIsVisible() {
        signInScreen().assertIsDisplayed()
    }
    
    fun assertHasRequiredElements() {
        tallyLogo().assertIsDisplayed()
        appTitle().assertIsDisplayed()
        tagline().assertIsDisplayed()
        signInButton().assertIsDisplayed()
        continueWithoutAccountButton().assertIsDisplayed()
    }
    
    fun assertIsSignedOut() {
        signInButton().assertIsDisplayed()
    }
}
